content/en/blog/2024/otel-generative-ai/index.md

```diff
@@ -0,0 +1,163 @@
+For generative AI, these conventions streamline monitoring, troubleshooting, and
+optimizing AI models by standardizing attributes such as model parameters,
+response metadata, and token usage. This consistency supports better
+observability across tools, environments, and APIs, helping organizations track
+performance, cost, and safety with ease.
+
+The
+[**Instrumentation Library**](https://opentelemetry.io/docs/specs/otel/overview/#instrumentation-libraries)
+is being developed within the
+[OpenTelemetry Python Contrib](https://github.com/open-telemetry/opentelemetry-python-contrib)
+project to automate telemetry collection for generative AI applications. The
+first release is a Python library for instrumenting OpenAI client calls, given
+Python's widespread use in AI development and the popularity of OpenAI. Designed
+to integrate seamlessly with OpenAI's API, this library captures spans and
+events, gathering essential data like model inputs, response metadata, and token
+usage in a structured format.
+
+## Key Signals for Generative AI
+
+The
+[Semantic Conventions for Generative AI](https://github.com/open-telemetry/semantic-conventions/tree/v1.28.0/docs/gen-ai)
+focus on capturing insights into AI model behavior through three primary
+signals: [Traces](https://opentelemetry.io/docs/concepts/signals/traces/),
+[Metrics](https://opentelemetry.io/docs/concepts/signals/metrics/), and
+[Events](https://opentelemetry.io/docs/specs/otel/logs/event-api/).
+
+Together, these signals provide a comprehensive monitoring framework, enabling
+better cost management, performance tuning, and request tracing.
+
+### Traces: Tracing Model Interactions
+
+Traces track each model interaction’s lifecycle, covering input parameters (for
+example, temperature, top_p) and response details like token count or errors.
+They provide visibility into each request, aiding in identifying bottlenecks and
+analyzing the impact of settings on model output.
+
+### Metrics: Monitoring Usage and Performance
+
+Metrics aggregate high-level indicators like request volume, latency, and token
+counts, essential for managing costs and performance. This data is particularly
+critical for API-dependent AI applications with rate limits and cost
+considerations.
+
+### Events: Capturing Detailed Interactions
+
+Events log detailed moments during model execution, such as user prompts and
+model responses, providing a granular view of model interactions. These insights
+are invaluable for debugging and optimizing AI applications where unexpected
+behaviors may arise.
+
+> Note that we decided to use the newer Events API
+> (https://opentelemetry.io/docs/specs/otel/logs/event-api/) specification in
+> the Semantic Conventions for Generative AI. The events API allows for us to
+> define specific
+> [semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/events/)
+> for the user prompts and model responses that we capture.
```

Comment:
```suggestion
{{% alert title="Note" color="info" %}}

Note that we decided to use the newer Events API
(https://opentelemetry.io/docs/specs/otel/logs/event-api/) specification in
the Semantic Conventions for Generative AI. The events API allows for us to
define specific
[semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/events/)
for the user prompts and model responses that we capture.

{{% /alert %}}
```

While the `>` accomplish something similar, let's use the [alert shortcode](https://www.docsy.dev/docs/adding-content/shortcodes/#alert)

Reply:
I feel strongly we should say "Log events API" as it is a subclass of logging, and if you don't configure logging properly nothing will work. This is particularly important because status quo before was span events.

Reply:
@codefromthecrypt I hear you, however I don't see it referred to that way elsewhere? Maybe it is, I just didn't see it.

Reply:
The [Events API](https://opentelemetry.io/docs/specs/otel/logs/event-api/) states that _"this existing Events API and once it is defined and implemented, the Events API usage will be migrated, deprecated, renamed and eventually removed."_ 

Reading this note (we use the new events api) and then reading that it is going to be deprecated, is confusing for an end user. 

@lmolkova @tedsuo any guidance here?

Reply:
@drewby it is in the log directory and many places it has the word log in the doc. We should fix the heading in the doc in my opinion if that's the only way to sort out this blog to highlight that this is log events not span events

Reply:
signal wise, we have log, metrics, spans, profile. If we want instead to literally say "events api" we should introduce it as a log signal. I personally jumping straight to "events api" doesn't prepare folks that this uses log signals, and a single word helps untangle this. EOF!

Reply:
Event API is being merged into Logs API - https://github.com/open-telemetry/opentelemetry-specification/blob/v1.39.0/specification/logs/api.md#emit-an-event.

So a future-proof way would be to say 

> we decided to use the log-based events

or

> we decided to use events emitted with Logs API

Reply:
I've updated the Alert text based on @lmolkova feedback.